lib/glific/contacts/import_worker.ex

```diff
@@ -0,0 +1,99 @@
+defmodule Glific.Contacts.ImportWorker do
+  @moduledoc """
+  Worker for processing contact chunks.
+  """
+  import Ecto.Query
+
+  require Logger
+
+  use Oban.Worker,
+    queue: :default,
+    max_attempts: 2,
+    priority: 2
+
+  alias Glific.{
+    Contacts.Import,
+    Jobs.UserJob,
+    Repo
+  }
+
+  @doc """
+  Creating new job for each chunk of contacts.
+  """
+  @spec make_job(list(), map(), non_neg_integer()) :: :ok
+  def make_job(chunk, params, user_job_id) do
+    __MODULE__.new(%{contacts: chunk, params: params, user_job_id: user_job_id})
+    |> Oban.insert()
+  end
+
+  @doc """
+  Standard perform method to use Oban worker.
+  """
+  @impl Oban.Worker
+  def perform(%Oban.Job{
+        args: %{"contacts" => contacts, "params" => params, "user_job_id" => user_job_id} = args
+      }) do
+
+    Repo.put_process_state(params["organization_id"])
+    result = Enum.reduce_while(contacts, %{}, fn contact, acc ->
+      phone_number = Map.get(contact, "phone")
+      acc = validate_phone(acc, phone_number)
+
+      if Map.has_key?(acc, :errors) do
+        {:halt, acc}
+      else
+        {:cont, acc}
+      end
+    end)
+
+    if Map.has_key?(result, :errors) do
+      {:error, result}
+    else
```

Comment:
There shouldn't be an if else case, even if we have errors we have to add it ot the user_job table and also increment the tasks done.